### ISSUE TITLE
Semitone/Cent units for Floaty Delay playrate? 

### ISSUE THREAD

**Issue by Andreya-Autumn** [surge-synthesizer/surge#7923]
Advantage: More familiarity. 
Disadvantage: May lead folks away from exploring other proportions. 

You use extend for a unit switch. But it doesn't actually change anything about how the slider works, so it seems a bit overkill. So probably let's think about this and decide on one for 1.4

**Comment by baconpaul** [comment]
There’s an alternate tooltip display in Param. That’s how the filter cutoff shows 440 and a4 on the left. Wonder if that would work here?

**Comment by baconpaul** [comment]
Would require a bit of plumbing and a pmd extension but would be doable

**Comment by mkruselj** [comment]
Yeah I suppose. I forgot this one is libized so it abides by different rules to how we used to do things in Surge codebase...

Also to add here, Jacky's request to be able to type in ratios and it converts to decimal, would be great to have.

**Comment by baconpaul** [comment]
So the param metadata already has a valueForAlternate method but right now it always returns `std::nullopt`

but since floaty types have types in surge you *could* do the alternate in Parameter.cpp.  Find `Parameter::get_display_alt` and then you can case it based on type in there since the valueForAlternate check will always fall through.

```
diff --git a/src/common/Parameter.cpp b/src/common/Parameter.cpp
index c6b38f72..bb89959f 100644
--- a/src/common/Parameter.cpp
+++ b/src/common/Parameter.cpp
@@ -3417,6 +3417,11 @@ void Parameter::get_display_alt(char *txt, bool external, float ef) const
         snprintf(txt, TXT_SIZE, "%s", bin.c_str());
         break;
     }
+    case ct_floaty_delay_playrate:
+    {
+        snprintf(txt, TXT_SIZE, "foo");
+        break;
+    }
     case ct_percent_bipolar_w_dynamic_unipolar_formatting:
     {
```

the type-in is harder. We need alternate inversion functions in the PMD which we dont' have right now.

So my advice for right now is

1: code up the alternate display in surge
2. don't do the alternate typeins
3. open an issue in basic blocks or leave this one open for the PMD extension required

**Comment by mkruselj** [comment]
I am going to close this one and open an issue in sst-basic-blocks. We can type in linear ratios for playrate, at least. The rest is all working, except semitone typeins.
